### DISCUSSION TITLE
How to test Webview with react-native-testing-library?

### DISCUSSION THREAD

**Discussion by fabiendem** [react-native-webview/react-native-webview#2689] [Q&A]
Hello,

I am struggling to find good examples of tests while using the `react-native-webview` and  `react-native-testing-library` under jest.

How do you usually test components which render a webview? Especially, how do you simulate the successful loading of the webpage in the context of the test renderer?

Thanks!

Edit: Asked as well at https://github.com/callstack/react-native-testing-library/issues/1255

**Comment by udaisshekhawat** [comment]
Anything on the above query please?

Even I am looking for some tests using @testing-library/react-native, such as validation onNavigationStateChange.

When I try to print the screen.debug(), the onNavigationStateChange is missing. It will be great if somebody knows anything on this.

Thanks

**Reply by fabiendem** [reply]
Hi, See https://github.com/callstack/react-native-testing-library/issues/1255#issuecomment-1342794075
